staging/src/k8s.io/client-go/tools/remotecommand/remotecommand.go

```diff
@@ -68,6 +68,29 @@ type streamExecutor struct {
 	protocols []string
 }
 
+// upgraderWithTerminateFunc wraps an upgrader with a terminate func.
+type upgraderWithTerminateFunc struct {
+	upgrader spdy.Upgrader
+	conns []httpstream.Connection
+}
+
+func (upgrader *upgraderWithTerminateFunc) NewConnection(resp *http.Response) (httpstream.Connection, error) {
+	conn, err := upgrader.upgrader.NewConnection(resp)
+	if err != nil {
+		return nil, err
+	}
+	upgrader.conns = append(upgrader.conns, conn)
+
+	return conn, nil
+}
+
+func (upgrader *upgraderWithTerminateFunc) terminate() {
+	// close all conns opened by this upgrader/tripper
+	for _, conn := range upgrader.conns {
+		_ = conn.Close()
```

Comment:
According to https://github.com/kubernetes/client-go/issues/884#issuecomment-718756465, closing the connections doesn't kill the exec itself, which may still cause memory leak. Do you plan on solving that problem?

Reply:
Yes. The exec process in the container will be still alive until it receive a exit signal. I am trying to find a solution and maybe it's more complex.